### ISSUE TITLE
Error when applying "assign_scalar_variable_to_conditions_process"

### ISSUE THREAD

**Issue by AlejandroCornejo** [KratosMultiphysics/Kratos#2419]
Now I'm currently running some cases with the StructuralMechApp using this process:
`        "python_module" : "assign_scalar_variable_to_conditions_process",
        "kratos_module" : "KratosMultiphysics",
        "Parameters"    : {
            "model_part_name" : "SurfacePressure3D_Pressure_on_surfaces_Auto1",
            "variable_name"   : "POSITIVE_FACE_PRESSURE",
            "value"           : 500,
            "interval"        : [0.0,"End"]`

and **I've noticed that when I use a value such as "500*t" you get this error:**

> RuntimeError: Error: Not able to set the variable. Attempting to set variable:POSITIVE_FACE_PRESSURE

in kratos/processes/assign_scalar_field_to_conditions_process.h:126:AssignScalarFieldToConditionsProcess::Execute
   kratos/processes/assign_scalar_field_to_conditions_process.h:129:AssignScalarFieldToConditionsProcess::Execute

The error is located here:
![imagen](https://user-images.githubusercontent.com/26402605/42177130-e55f2ebc-7e2b-11e8-8075-c0c547b6f22f.png)

How can I solve this error?

Thank you,

Alejandro
@loumalouomega @RiccardoRossi

**Comment by loumalouomega** [comment]
This is because POSITIVE_FACE_PRESSURE is registered as a Vector, not as an array_1d<double, 3>, can be solved easily with a minor change

**Comment by AlejandroCornejo** [comment]
Then i do not understand why it's working with a constant value of the load...

**Comment by loumalouomega** [comment]
Probably because we are ****, look at the code, where is getting adouble is getting a vector:

~~~ cpp
if( KratosComponents< Variable<Vector> >::Has( mVariableName ) ) //case of double variable
~~~

**Comment by loumalouomega** [comment]
Certainly I need to refactor this, and I don't understand how it was working until now

**Comment by philbucher** [comment]
@navaneethkn I think you had the same problem right?

**Comment by loumalouomega** [comment]
Ok, the problems was in the comments, the variable is actually a Vector, not a  double

**Comment by AlejandroCornejo** [comment]
so you are saying that we should register the variable  as a double?

**Comment by loumalouomega** [comment]
Nope, the comment was wrong and there is an error in the implementation, one sec

**Comment by loumalouomega** [comment]
![imagen](https://user-images.githubusercontent.com/19991680/42184454-dfb94e60-7e45-11e8-8c95-cdcba168ed35.png)

**Comment by loumalouomega** [comment]
Maybe I was right the first time, now I see that this process is jhust called for scalr variables...

**Comment by loumalouomega** [comment]
There is a big confussion, the variables are stored in vector because the size of the vector is the number of nodes because is computed as a vector of N components

**Comment by loumalouomega** [comment]
The variable should be a double,a and the assign is not correct, the process was working because two bad implementation made half functional one

**Comment by loumalouomega** [comment]
My suggestion is to change the variables declared as Vector to Components, if it has 3 components of course

**Comment by loumalouomega** [comment]
The use of the FACE_NEGATIVE_POSITIVE pressure is terribly inconsistent. When is assigned to a condition is used as a double, when is assigned to nodes is used as Vector

https://github.com/KratosMultiphysics/Kratos/blob/0fe8624e385cd1ef37041c87a92d47ffa131439f/applications/StructuralMechanicsApplication/custom_conditions/surface_load_condition_3d.cpp#L240

We need to refactor all this, or it will bring to confusion and severe errors in the future.

**Comment by RiccardoRossi** [comment]
Hi Guys,

the problem is that if you want to apply a scalar field on a condition, than you must provide a vector of size the number of nodes so to be able to compute the values on each of the nodes.

the problem here is that the field can be discontinuous condition-to-condition (think to a snow load)

please DO NOT make changes to this in a rush.

I am fine with discussing it and in changing it, but ...first the discussion ... THEN the changes

**Comment by loumalouomega** [comment]
I changed the title of the PR to do not merge.

Besides, the test added brings to confusion, the variable used is the INITIAL_STRAIN, which is a vector because it is a strain, not because is assigned node to node, this brings to confusion.

Additionally the implementation in the structural is wrong if the idea is to use a vector to assign node to node. Look my previous comment.

**Comment by navaneethkn** [comment]
I had this problem when I was using POSITIVE_FACE_PRESSURE in my branch. I fixed it by adding an additional if condition. If you think it is a right fix then I can raise the pull request.

![image](https://user-images.githubusercontent.com/18971874/42209098-1bee5642-7eae-11e8-95f3-0a4b594497b8.png)

**Comment by loumalouomega** [comment]
Yes, this was missing, but with the lack of explanation i assumed I was supposed to be double by default and the Vector was a mistake, for the comment I assumed

**Comment by loumalouomega** [comment]
That is now solved on #2420

**Comment by loumalouomega** [comment]
In any case we should discuss about this
